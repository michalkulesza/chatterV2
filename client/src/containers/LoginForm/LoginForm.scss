@use '../../styles/global';

.LoginForm {
	background-color: global.$white;
	padding: 3rem 4.5rem;
	border-radius: 10px;
	box-shadow: global.$shadow;
	width: 80%;
	max-width: 450px;

	.Container {
		display: flex;
		justify-content: center;
		flex-direction: column;
		text-align: center;

		.Logo {
			font-family: "Maven Pro", sans-serif;
			font-weight: 700;
			font-size: 2rem;
			color: global.$gray;
		}

		.Title {
			font-size: 2.5rem;
			font-weight: 600;
			margin-bottom: 2rem;
		}
	}

	form {
		display: flex;
		flex-direction: column;
		gap: 1.5rem;
	}
}
