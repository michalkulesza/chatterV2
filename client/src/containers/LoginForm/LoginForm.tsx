import React, { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { loginWithoutPassword } from "../../redux/actions/auth";
import { addAuthError } from "../../redux/actions/error";
import { RootState } from "../../redux/reducers/rootReducer";
import "./LoginForm.scss";

import FormInput from "./FormInput/FormInput";
import FormButton from "./FormButton/FormButton";

interface Props {}

const LoginForm: React.FC<Props> = () => {
	const dispatch = useDispatch();
	const error = useSelector((state: RootState) => state.error.auth);
	const [user, setUser] = useState("");
	const [password, setPassword] = useState("");
	const [register, setRegister] = useState(false);

	const hasLength = (string: string, length: number) => string.length >= length;

	const validName = (name: string) => hasLength(name, 3) && name !== "admin";

	const validPassword = (password: string) => hasLength(password, 4);

	const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
		e.preventDefault();
		const passwordEntered = password.length > 0;

		if (!validName(user)) return dispatch(addAuthError("Username must be at least 3 characters long."));

		if (!validPassword(password) && (register || passwordEntered))
			return dispatch(addAuthError("Password must be at least 4 characters long."));

		if (passwordEntered && register) return console.log("REGISTER");
		if (passwordEntered) return console.log("LOGIN WITHOUT PASSWORD");
		dispatch(loginWithoutPassword(user));
	};

	return (
		<div className="LoginForm">
			<div className="Container">
				<div className="Logo">Chatter</div>
				<div className="Title">Join In</div>
				<form onSubmit={e => handleSubmit(e)}>
					<FormInput type="text" setState={setUser} state={user} label="Username" required />
					<FormInput
						type="password"
						setState={setPassword}
						state={password}
						label="Password (optional)"
						required={register}
					/>
					<FormInput type="checkbox" setState={setRegister} state={register} label="I want to register Username" />
					{error && <div className="error">{error}</div>}
					<FormButton>Join</FormButton>
				</form>
			</div>
		</div>
	);
};

export default LoginForm;
