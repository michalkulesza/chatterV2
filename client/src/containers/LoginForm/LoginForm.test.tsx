import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureMockStore from "redux-mock-store";
import LoginForm from "./LoginForm";

const initState = {
	error: {
		auth: null,
	},
	auth: {
		loading: false,
	},
};

const middlewares = [thunk];
const store = configureMockStore(middlewares)(initState);

describe("LoginForm", () => {
	it("Renders LoginForm properly", () => {
		const wrapper = mount(
			<Provider store={store}>
				<LoginForm />
			</Provider>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it("Renders inputs", () => {
		const wrapper = mount(
			<Provider store={store}>
				<LoginForm />
			</Provider>
		);

		expect(wrapper.find("input").length).toBe(3);
	});
});
