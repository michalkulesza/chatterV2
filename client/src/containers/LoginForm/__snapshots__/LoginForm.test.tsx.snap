// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Renders LoginForm properly 1`] = `ReactWrapper {}`;
